velox/common/file/FileSystems.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <functional>
+#include <string_view>
+
+#define _VELOX_REGISTER_FILE_SYSTEM(tag) registerFileSystem_##tag
```

Comment:
Why do we need this? How does this work?

Reply:
I followed the `VELOX_REGISTER_VECTOR_FUNCTION` pattern here. See `VectorFunction.h`.
I am going to write more details in the PR description.

Reply:
That would work, but, perhaps, we can avoid these macros altogether. See velox/functions/sparksql/Register.cpp

Reply:
The pragmas can be avoided here. I will update it.